service/rpc/servicer.go

```diff
@@ -529,3 +843,22 @@ func (s *Servicer) blockHeader(ctx context.Context, blockIdentifier *types.Parti
 	}
 	return blockHeader, nil
 }
+
+func buildCallOpsFromCallRequest(ctx context.Context,
+	callRequest *types.CallRequest) (*bind.CallOpts, error){
+	if callRequest.Parameters["blockNumber"] == nil {
```

Comment:
Curious why you do this for `blockNumber`, but not the other common parameters like `address` or `group`?

Reply:
yeah, I should do for all. Mainly, there was more code for block number because it's ok if it's nil. Whereas the others fail if nil.

Reply:
For the others, it doesn't save much space, as it's still `call, if error, return error`